components/base/kinematicbase/ptgKinematics.go

```diff
@@ -157,21 +159,28 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
 		linVel := r3.Vector{0, trajNode.LinVelMMPS, 0}
 		angVel := r3.Vector{0, 0, rdkutils.RadToDeg(trajNode.AngVelRPS)}
 
-		ptgk.logger.Debugf(
-			"setting velocity to linear %v angular %v and running velocity step for %s",
-			linVel,
-			angVel,
-			timestep,
-		)
-
-		err := ptgk.Base.SetVelocity(
-			ctx,
-			linVel,
-			angVel,
-			nil,
-		)
-		if err != nil {
-			return multierr.Combine(err, ptgk.Base.Stop(ctx, nil))
+		// This should call SetVelocity if:
+		// 1) this is the first iteration of the loop, or
+		// 2) either of the linear or angular velocities has changed
+		if i == 0 || !(linVel.ApproxEqual(lastLinVel) && angVel.ApproxEqual(lastAngVel)) {
```

Comment:
could get rid of the need for the first case if you initialize the last vectors to something other than the zero vector

Reply:
I thought about initializing to `math.Inf(1)` repeated three times but I think this is more readable.

Reply:
And there are no other values that we can guarantee will never be called.

Also considered making it a pointer with a nil check, but that's more than keeping track of the loop var that we already have.

Reply:
I like the infinities personally, keeps you from having to do this logic every iteration. But super minor so whatever you decide to do is good with me

Reply:
Branch prediction means that this extra conditional will be a no-op.

Reply:
I prefer the index test as it makes it clear it is just testing for which iteration of the loop you are on.